tests/cache_scalar.rs

```diff
@@ -108,3 +112,168 @@ async fn keys_exist_happy_path() -> MomentoResult<()> {
+    assert_eq!(result.value, 1);
+
+    // Incrementing an existing key should increment it
+    let result = client.increment(cache_name, &**key, 1).await?;
+    assert_eq!(result.value, 2);
+
+    // Incrementing by a negative number should decrement the value
+    let result = client.increment(cache_name, &**key, -2).await?;
+    assert_eq!(result.value, 0);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn item_get_type_invalid_cache_name() -> MomentoResult<()> {
+    let client = &CACHE_TEST_STATE.client;
+    let result = client.item_get_type("   ", "key").await.unwrap_err();
+    assert_eq!(result.error_code, MomentoErrorCode::InvalidArgumentError);
+    Ok(())
+}
+
+#[tokio::test]
+async fn item_get_type_nonexistent_cache() -> MomentoResult<()> {
+    let client = &CACHE_TEST_STATE.client;
+    let cache_name = "fake-cache-".to_string() + &Uuid::new_v4().to_string();
+    let result = client.item_get_type(cache_name, "key").await.unwrap_err();
+    assert_eq!(result.error_code, MomentoErrorCode::NotFoundError);
+    Ok(())
+}
+
+#[tokio::test]
+async fn item_get_type_happy_path() -> MomentoResult<()> {
+    let client = &CACHE_TEST_STATE.client;
+    let cache_name = &CACHE_TEST_STATE.cache_name;
+    let key = &Uuid::new_v4().to_string();
+
+    // Expect miss when key is not set
+    let result = client.item_get_type(cache_name, &**key).await?;
+    assert_eq!(result, ItemGetType::Miss {});
+    client.delete(cache_name, &**key).await?;
+
+    // Expect Scalar after using set
+    client.set(cache_name, &**key, "value").await?;
+    let result = client.item_get_type(cache_name, &**key).await?;
+    match result {
+        ItemGetType::Hit { key_type } => assert_eq!(
+            key_type,
+            ItemType::Scalar,
+            "Expected Scalar, got {:?}",
+            key_type
+        ),
+        _ => panic!("Expected Hit, got {:?}", result),
+    }
+    client.delete(cache_name, &**key).await?;
+
+    // Expect Set after using setAddElements
+    client
+        .set_add_elements(cache_name, &**key, vec!["value1", "value2"])
+        .await?;
+    let result = client.item_get_type(cache_name, &**key).await?;
+    match result {
+        ItemGetType::Hit { key_type } => {
+            assert_eq!(key_type, ItemType::Set, "Expected Set, got {:?}", key_type)
+        }
+        _ => panic!("Expected Hit, got {:?}", result),
+    }
+    client.delete(cache_name, &**key).await?;
+
+    // Expect SortedSet after using sortedSetPutElements
+    client
+        .sorted_set_put_elements(cache_name, &**key, vec![("value1", 1.0), ("value2", 2.0)])
+        .await?;
+    let result = client.item_get_type(cache_name, &**key).await?;
+    match result {
+        ItemGetType::Hit { key_type } => assert_eq!(
+            key_type,
+            ItemType::SortedSet,
+            "Expected SortedSet, got {:?}",
+            key_type
+        ),
+        _ => panic!("Expected Hit, got {:?}", result),
+    }
+    client.delete(cache_name, &**key).await?;
```

Comment:
oh this is interesting, we should make sure we remember to come back and add Dictionary/List here once those exist.